java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java

```diff
@@ -260,6 +260,6 @@ public BufferAllocator getAllocator() {
 
   @Override
   public boolean accept(RangeEqualsVisitor visitor) {
-    return visitor.visit(getUnderlyingVector());
+    return visitor.visitGeneral(getUnderlyingVector());
```

Comment:
can we instead do -

return getUnderlyingVector().accept(visitor)

Reply:
I think this is also ok, and then should remove RangeEqualsVisitor#visit(ValueVector left, Void value).

Reply:
Sorry for correction, if we don't have visitGeneral I think the API below will throw exception

>public boolean equals(ValueVector left) {
    return this.visit(left, null);
  }

Reply:
The method suggested by @pravindra works. With the suggested change, method visit(ValueVector left) can be removed altogether.